servicetests/tests/ShadowUpdate/main.cpp

```diff
@@ -0,0 +1,352 @@
+     * In a real world application you probably don't want to enforce synchronous behavior
+     * but this is a sample console application, so we'll just do that with a condition variable.
+     */
+    std::promise<bool> connectionCompletedPromise;
+    std::promise<void> connectionClosedPromise;
+    // Create connection
+    std::shared_ptr<Aws::Crt::Mqtt::MqttConnection> connection = nullptr;
+    // Create Mqtt5Client
+    std::shared_ptr<Aws::Crt::Mqtt5::Mqtt5Client> client5 = nullptr;
+
+    if (cmdData.input_mqtt_version == 5UL)
+    {
+        shadowClient = build_mqtt5_client(cmdData, client5, connectionCompletedPromise, connectionClosedPromise);
+    }
+    else if (cmdData.input_mqtt_version == 3UL)
+    {
+        shadowClient = build_mqtt3_client(cmdData, connection, connectionCompletedPromise, connectionClosedPromise);
+    }
+    else
+    {
+        fprintf(stderr, "MQTT Version not supported\n");
+        exit(-1);
+    }
+
+    /************************ Run the sample ****************************/
+    if (connectionCompletedPromise.get_future().get())
+    {
+        if (cmdData.input_shadowName.empty())
+        {
+            changeShadowValue(
+                cmdData.input_thingName, cmdData.input_shadowProperty, cmdData.input_shadowValue, shadowClient);
+        }
+        else
+        {
+            changeNamedShadowValue(
+                cmdData.input_thingName,
+                cmdData.input_shadowProperty,
+                cmdData.input_shadowValue,
+                cmdData.input_shadowName,
+                shadowClient);
+        }
+    }
+    /************************ sample ends ****************************/
+    /* Closing down */
+    // Wait just a little bit to let the console print
+    std::this_thread::sleep_for(std::chrono::milliseconds(500));
+
+    if (cmdData.input_mqtt_version == 5UL)
+    {
+        // Disconnect
+        if (client5->Stop() == true)
+        {
+            connectionClosedPromise.get_future().wait();
+        }
+    }
+    else
+    { // mqtt3
+
+        // Disconnect
+        if (connection->Disconnect() == true)
+        {
+            connectionClosedPromise.get_future().wait();
+        }
+    }
+    return 0;
+}
+
+void changeShadowValue(
+    Aws::Crt::String thingName,
+    String property,
+    String value,
+    std::shared_ptr<IotShadowClient> shadowClient)
+{
+    printf("change shadow value ....\n");
+    JsonObject desired;
+    JsonObject reported;
+    ShadowState state;
+
+    JsonObject val;
+    val.AsString(value);
+
+    desired.WithObject(property, val);
+    reported.WithObject(property, val);
+
+    state.Desired = desired;
+    state.Reported = reported;
+
+    UpdateShadowRequest request;
+    request.ThingName = thingName;
+    request.State = state;
+    Aws::Crt::UUID uuid;
+    request.ClientToken = uuid.ToString();
+
+    std::promise<void> shadowCompletedPromise;
+    auto publishCompleted = [thingName, value, &shadowCompletedPromise](int ioErr) {
```

Comment:
We're still not verifying anything actually works beyond connecting.

Reply:
true, because we are verifying in the launch script, that the job did execute